torch/csrc/jit/passes/memory_planning.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include <c10/core/ScalarType.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/memory_planning.h>
+#include <torch/csrc/jit/runtime/static/impl.h>
+
+namespace torch {
+namespace jit {
+
+c10::TensorTypePtr checkIsSupported(const Value& value) {
+  TORCH_CHECK(
+      value.type()->cast<TensorType>(),
+      "out isn't a tensortype ",
+      *value.type());
+  auto ttp = value.type()->expect<TensorType>();
+  TORCH_CHECK(
+      ttp->scalarType().has_value(),
+      "This output was profiled but didn't have a scalar type.")
+  TORCH_CHECK(
+      ttp->sizes().concrete_sizes().has_value(),
+      "This output was profiled but doesn't have sizes.")
+  return ttp;
+}
+
+std::pair<std::vector<int64_t>, std::vector<int64_t>> getSizesStrides(
+    const c10::TensorTypePtr& ttp) {
+  auto sizes = ttp->sizes().concrete_sizes().value();
+  std::vector<int64_t> strides;
+  auto _strides = ttp->strides().concrete_sizes();
+  if (_strides.has_value() && _strides.value()[0] != 0) {
+    strides = _strides.value();
+  } else {
+    strides = at::detail::defaultStrides(sizes);
+  }
+  return std::make_pair(sizes, strides);
+}
+
+size_t computeStorageSize(const c10::TensorTypePtr& ttp) {
+  std::vector<int64_t> sizes, strides;
+  std::tie(sizes, strides) = getSizesStrides(ttp);
+  return at::detail::computeStorageNbytes(
+      sizes, strides, elementSize(ttp->scalarType().value()));
+}
+
+std::pair<size_t, std::map<Value*, c10::TensorTypePtr>>
+computeAlignedStorageSize(std::vector<Node*>& nodes) {
+  // TODO: liveness analysis
```

Comment:
👍

Reply:
working on it as we speak 😂

Reply:
@desertfire oh great! just curious: is there an explanation somewhere on the technique?

Reply:
we should be extremely careful reusing code from previous projects without auditing them, bailouts/static runtime etc. the liveness linked there only does Value liveness it does not do alias analysis / handling of views etc